modules/core/arrow-core-data/src/test/kotlin/arrow/core/extensions/NumberSemiringTest.kt

```diff
@@ -1,80 +1,23 @@
-        testLaws(SemiringLaws.laws(Double.semiring(), A.toDouble(), B.toDouble(), C.toDouble(), Double.eq()))
-        testLaws(SemiringLaws.laws(Int.semiring(), A, B, C, Int.eq()))
-        testLaws(SemiringLaws.laws(Short.semiring(), A.toShort(), B.toShort(), C.toShort(), Short.eq()))
-        testLaws(SemiringLaws.laws(Float.semiring(), A.toFloat(), B.toFloat(), C.toFloat(), Float.eq()))
-
-        /**
-         * maybeCombineMultiplicate()
-         */
-
-        "maybeCombineMultiplicate() should return one() when called on null receiver" {
-            Int.semiring().run {
-                null.maybeCombineMultiplicate(1).shouldBe(Int.semiring().one())
-            }
-        }
-
-        "maybeCombineMultiplicate() should calculate 6 when called on 2 and 3" {
-            Int.semiring().run {
-                2.maybeCombineMultiplicate(3).shouldBe(6)
-            }
-        }
-
-        "maybeCombineMultiplicate() should return receiver value when argument is null" {
-            Int.semiring().run {
-                val receiverValue = 2
-                receiverValue.maybeCombineMultiplicate(null).shouldBe(receiverValue)
-            }
-        }
-
-        "maybeCombineMultiplicate() should return one() when receiver and argument are both null" {
-            Int.semiring().run {
-                null.maybeCombineMultiplicate(null).shouldBe(Int.semiring().one())
-            }
-        }
-
-        /**
-         * maybeCombineAddition()
-         */
-
-        "maybeCombineAddition() should return zero() when called on null receiver" {
-            Int.semiring().run {
-                null.maybeCombineAddition(1).shouldBe(Int.semiring().zero())
-            }
-        }
-
-        "maybeCombineAddition() should calculate 3 when called on 1 and 2" {
-            Int.semiring().run {
-                1.maybeCombineAddition(2).shouldBe(3)
-            }
-        }
-
-        "maybeCombineAddition() should return receiver value when argument is null" {
-            Int.semiring().run {
-                val receiverValue = 2
-                receiverValue.maybeCombineAddition(null).shouldBe(receiverValue)
-            }
-        }
-
-        "maybeCombineAddition() should return zero() when receiver and argument are both null" {
-            Int.semiring().run {
-                null.maybeCombineAddition(null).shouldBe(Int.semiring().zero())
-            }
-        }
-    }
+  init {
+    testLaws(SemiringLaws.laws(Byte.semiring(), Gen.byteSmall(), Byte.eq()))
+    testLaws(SemiringLaws.laws(Double.semiring(), Gen.doubleSmall(), Double.eq()))
+    testLaws(SemiringLaws.laws(Int.semiring(), Gen.intSmall(), Int.eq()))
+    testLaws(SemiringLaws.laws(Short.semiring(), Gen.shortSmall(), Short.eq()))
+    testLaws(SemiringLaws.laws(Float.semiring(), Gen.floatSmall(), Float.eq()))
+    testLaws(SemiringLaws.laws(Long.semiring(), Gen.longSmall(), Long.eq()))
```

Comment:
Hmm can you double check if all these are run? Iirc kotlintest filters out tests with same names (or was that just on our side?)

Reply:
Yep they are, because each `testLaws` call is doing the filtering